sdl_android/src/main/java/com/smartdevicelink/api/lockscreen/LockScreenManager.java

```diff
@@ -0,0 +1,285 @@
+					OEMIconUrl = msg.getUrl();
+					downloadLockScreenIcon(OEMIconUrl);
+				}
+			}
+		};
+		internalInterface.addOnRPCNotificationListener(FunctionID.ON_SYSTEM_REQUEST, systemRequestListener);
+	}
+
+	////
+	// LAUNCH LOCK SCREEN LOGIC
+	////
+
+	/**
+	 * 1. Check if user wants us to manage lock screen
+	 * 2. If so, get the HMI level and LockScreenStatus from the method below
+	 * 3. Build intent and start the SDLLockScreenActivity
+	 *
+	 * X. If the status is set to OFF, Send broadcast to close lock screen if it is open
+	 */
+	private void launchLockScreenActivity(){
+		// intent to open SDLLockScreenActivity
+		// pass in icon, background color, and custom view
+		if (lockScreenEnabled) {
+			LockScreenStatus status = getLockScreenStatus();
+			if (hmiLevel == HMILevel.HMI_FULL && status == LockScreenStatus.REQUIRED) {
+				Intent showLockScreenIntent = new Intent(context.get(), SDLLockScreenActivity.class);
+				showLockScreenIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+				// Extra parameters for customization of the lock screen view
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_ICON_EXTRA, lockScreenIcon);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_COLOR_EXTRA, lockscreenColor);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_CUSTOM_VIEW_EXTRA, customView);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_OEM_ICON_EXTRA, showOEMLogo);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_OEM_ICON_BITMAP, lockScreenOEMIcon);
+
+				context.get().startActivity(showLockScreenIntent);
+			} else if (status == LockScreenStatus.OFF) {
+				context.get().sendBroadcast(new Intent(SDLLockScreenActivity.CLOSE_LOCK_SCREEN_ACTION));
+			}
+		}
+	}
+
+	////
+	// HELPERS
+	////
+
+	/**
+	 * Step through some logic to determine if we need to show the lock screen or not
+	 * This function is usually triggered on some sort of notification.
+	 *
+	 * @return Whether or not the Lock Screen is required
+	 */
+	private synchronized LockScreenStatus getLockScreenStatus() {
+
+		if ( (hmiLevel == null) || (hmiLevel.equals(HMILevel.HMI_NONE))) {
+			return LockScreenStatus.OFF;
+		}
+		else if ( hmiLevel.equals(HMILevel.HMI_BACKGROUND)) {
+			if (!driverDistStatus) {
+				//we don't have driver distraction, lock screen is entirely based on if user is using the app on the head unit
+				if (isAppInUse) {
+					return LockScreenStatus.REQUIRED;
+				} else {
+					return LockScreenStatus.OFF;
+				}
+			}
+			else if (isAppInUse) {
+				return LockScreenStatus.REQUIRED;
+			} else {
+				return LockScreenStatus.OFF;
+			}
+		}
+		else if ( (hmiLevel.equals(HMILevel.HMI_FULL)) || (hmiLevel.equals(HMILevel.HMI_LIMITED))) {
+			if (!driverDistStatus) {
+				return LockScreenStatus.OPTIONAL;
+			} else {
+				return LockScreenStatus.REQUIRED;
+			}
+		}
+		return LockScreenStatus.OFF;
+	}
+
+	private void downloadLockScreenIcon(final String url){
```

Comment:
just to put down what we had discussed in person, the reason I chose to place this here rather than the activity is because there is no good way to hold onto this image as the activity is instantiated and killed frequently. If there was a race condition where `showOEMLogo` is true and the lockscreen is instantiated before the download completes, it will not show the logo the first time, but will show it every subsequent time for the duration of the session. This prevents unnecessary repeated downloads for what i believe is a small trade off. This alone with only downloading if wanted (your comment from above), I believe this is pretty good

Reply:
Why not just add a ping to the Activity that the image has been dowloaded and is ready for retrieval? The activity already has a broadcast receiver so we should be able to simply notify that with an intent.

Reply:
generalized the broadcast receiver and added this. done.